content/en-US/api/URLPattern.md

```diff
@@ -0,0 +1,515 @@
+## Properties
+
+**`URLPattern.protocol`**
+
+Returns the URL protocol pattern set during construction.  This value may
+be differ from the input to the constructor due to normalization.
+
+**`URLPattern.username`**
+
+Returns the URL username pattern set during construction.  This value may
+be differ from the input to the constructor due to normalization.
+
+**`URLPattern.password`**
+
+Returns the URL username pattern set during construction.  This value may
+be differ from the input to the constructor due to normalization.
+
+**`URLPattern.hostname`**
+
+Returns the URL username pattern set during construction.  This value may
+be differ from the input to the constructor due to normalization.
+
+**`URLPattern.port`**
+
+Returns the URL username pattern set during construction.  This value may
+be differ from the input to the constructor due to normalization.
+
+**`URLPattern.pathname`**
+
+Returns the URL username pattern set during construction.  This value may
+be differ from the input to the constructor due to normalization.
+
+**`URLPattern.search`**
+
+Returns the URL username pattern set during construction.  This value may
+be differ from the input to the constructor due to normalization.
+
+**`URLPattern.hash`**
+
+Returns the URL username pattern set during construction.  This value may
+be differ from the input to the constructor due to normalization.
+
+## Events
+
+None
+
+## Methods
+
+**`URLPattern.test()`**
+
+Matches the input against the pattern returning true on success and false
+otherwise.
+
+**`URLPattern.exec()`**
+
+Matches the input against the pattern returning a URLPatternResult object on
+success and null otherwise.
+
+## Examples
+
+### Example: Construct a URLPattern to filter on a specific URL component.
+
+The following example shows how a URLPattern can be used to filter on a
+specific URL component.  When the URLPattern constructor is called with a
+structured object of component patterns any missing components default to
+the `*` wildcard value.
+
+```js
+// Construct a URLPattern that matches a specific domain and its subdomains.
+// All other URL components default to the wildcard `*` pattern.
+const pattern = new URLPattern({
+  hostname: {*.}?example.com
+});
+
+// `{*.}?example.com`
+pattern.hostname;
+
+// `*`
+pattern.protocol;
+pattern.username;
+pattern.password;
+pattern.pathname;
+pattern.search;
+pattern.hash;
+
+// true
+pattern.test("https://example.com/foo/bar");
+
+// true
+pattern.test({ hostname: "cdn.example.com" });
+
+// true
+pattern.test("custom-protocol://example.com/other/path?q=1");
+
+// false because the hostname component does not match
+pattern.test("https://cdn-example.com/foo/bar");
+```
+
+### Example: Construct a URLPattern from a full URL string.
+
+The following example shows how a URLPattern can be constructed from a full
+URL string with patterns embedded.  If there is no ambiguity between whether
+a character is a URL separate or a pattern character then it just works.
```

Comment:
What are URL separates and pattern characters?

Reply:
That should be URL "separator".  Like how `https://example.com/" has a colon after the "https" protocol.  Pattern strings also use colon for the beginning of named groups like ":foo".  Another example of this is `?` which is used before URL search params like `https://example.com/?q=1` and in pattern strings as an optional modifier like `:foo?`.

Reply:
I tried to reword this to be more clear in the latest push.